cli/sdo/voucher.go

```diff
@@ -151,6 +151,126 @@ type ImportResponse struct {
+	msgPrinter := i18n.GetMessagePrinter()
+	cliutils.Verbose(msgPrinter.Sprintf("Listing imported SDO vouchers."))
+
+	var output interface{}
+	err := json.Unmarshal(respBodyBytes, &output)
+	if err != nil {
+		cliutils.Fatal(cliutils.HTTP_ERROR, msgPrinter.Sprintf("json unmarshalling HTTP response '%s' from %s: %v", string(respBodyBytes), apiMsg, err))
+	}
+
+	jsonBytes, err := json.MarshalIndent(output, "", cliutils.JSON_INDENT)
+	if err != nil {
+		cliutils.Fatal(cliutils.JSON_PARSING_ERROR, msgPrinter.Sprintf("failed to marshal 'hzn exchange service list' output: %v", err))
+	}
+
+	return jsonBytes
+}
+
+// list the all the uploaded SDO vouchers, or a single voucher
+func VoucherList(org, userCreds, voucher string, namesOnly bool) {
+	msgPrinter := i18n.GetMessagePrinter()
+	cliutils.Verbose(msgPrinter.Sprintf("Listing imported SDO vouchers."))
+
+	// call the ocs-api to get the uploaded vouchers
+	var respBodyBytes []byte
+	var apiMsg string
+	respBodyBytes, apiMsg = getVouchers(org, userCreds, apiMsg, voucher)
+
+	// no voucher was specified, list all uploaded vouchers
+	if voucher == "" {
+		output := []string{}
+		err := json.Unmarshal(respBodyBytes, &output)
+		if err != nil {
+			cliutils.Fatal(cliutils.HTTP_ERROR, msgPrinter.Sprintf("json unmarshalling HTTP response '%s' from %s: %v", string(respBodyBytes), apiMsg, err))
+		}
+
+		jsonBytes, err := json.MarshalIndent(output, "", cliutils.JSON_INDENT)
+		if err != nil {
+			cliutils.Fatal(cliutils.JSON_PARSING_ERROR, msgPrinter.Sprintf("failed to marshal 'hzn exchange service list' output: %v", err))
+		}
+
+		// list only the uuid's of imported vouchers
+		if namesOnly {
+			fmt.Printf("%s\n", jsonBytes)
+
+		} else { // list full details of all imported vouchers
+			for i := range output {
+				respBodyBytes, apiMsg = getVouchers(org, userCreds, apiMsg, output[i])
+				jsonBytes := listFullVoucher(respBodyBytes, apiMsg)
+				fmt.Printf("%s\n", jsonBytes)
+				fmt.Printf("\n")
+			}
+		}
+
+	} else { // list the details of a single imported voucher 
+		vouch := InspectOutput{}
+		err := json.Unmarshal(respBodyBytes, &vouch)
+		if err != nil {
+			cliutils.Fatal(cliutils.HTTP_ERROR, msgPrinter.Sprintf("json unmarshalling HTTP response '%s' from %s: %v", string(respBodyBytes), apiMsg, err))
+		}
+		if namesOnly { // list only short relavent details of voucher
+			if err = parseVoucherBytes(respBodyBytes, &vouch); err != nil {
+				cliutils.Fatal(cliutils.JSON_PARSING_ERROR, msgPrinter.Sprintf("parsing the json from %s: %v", voucher, err))
+			}
+	
+			jsonBytes, err := json.MarshalIndent(vouch, "", cliutils.JSON_INDENT)
+			if err != nil {
+				cliutils.Fatal(cliutils.JSON_PARSING_ERROR, msgPrinter.Sprintf("failed to marshal 'hzn exchange service list' output: %v", err))
+			}
+			fmt.Printf("%s\n", jsonBytes)
+
+		} else { // list full details of voucher
+			jsonBytes := listFullVoucher(respBodyBytes, apiMsg)
+			fmt.Printf("%s\n", jsonBytes)
+		}
+	}
+}
+
```

Comment:
I don't think Ling will let you get away with merging this PR w/o the corresponding tests in `test/gov/hzn_voucher.sh`. But maybe she'll be nice and let you do it in a subsequent PR, since this is your 1st hzn PR :)

Reply:
Ah well hopefully! I'll start looking into this file now though